flycheck.el

```diff
@@ -8711,6 +8715,25 @@ rustc command.  See URL `https://www.rust-lang.org'."
                (and (flycheck-buffer-saved-p)
                     (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
 
+(flycheck-define-checker rust-cargo-clippy
+  "A Rust syntax checker and linter using clippy.
+
+This syntax checker needs the cargo clippy subcommand. See URL `https://github.com/Manishearth/rust-clippy'"
+  :command ("cargo" "clippy"
+            (eval flycheck-cargo-clippy-args)
+            "--"  "--error-format=json"
+            (option-list "-L" flycheck-rust-library-path concat)
+            (eval flycheck-rust-args))
+  :error-parser flycheck-parse-rust
+  :error-explainer flycheck-rust-error-explainer
+  :modes rust-mode
+  :predicate (lambda ()
+               ;; Since we build the entire project with cargo clippy we require
+               ;; that the buffer is saved.  And of course, we also need a Cargo
+               ;; file :)
+               (and (flycheck-buffer-saved-p)
+                    (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
```

Comment:
By the way, we'd also need a `:verify` function that informs the user about whether `Cargo.toml` was found or not, to help them figure out why a particular syntax checker isn't available in `C-c ! v`.  Take a look at `javascript-eslint` for an example.